### PR TITLE
Update 1.6.2

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ehelms** [comment]
This rebases onto 1.6.2 which is latest 1.x stable.

**Comment by ehelms** [comment]
This will need to be push directly to local eventually. The testing strategy is:

 1) git clone and have this branch
 2) pip install . --upgrade
 3) git clone foreman-infra
 4) cd puppet/modules/jenkins_job_builder/files/theforeman.org
 5) ./generate_jobs

We should try to figure out getting these three commits into JJB properly though. As requiring to have this fork installed can be annoying.

**Comment by mmoll** [comment]
rebase/merge needed?

File: jenkins_jobs/modules/publishers.py [hunk 1]
```diff
@@ -102,6 +102,11 @@ def archive(parser, xml_parent, data):
         success = XML.SubElement(archiver, 'onlyIfSuccessful')
         success.text = str(data.get('only-if-success', False)).lower()
 
+    if 'only-successful' in data:
```
Comment (ekohl) [inline jenkins_jobs/modules/publishers.py:105]: Current master has ```only-if-success```: https://github.com/openstack-infra/jenkins-job-builder/blob/f2b01f7c5ca29f6cf3fa1040106ed9d81fad32af/jenkins_jobs/modules/publishers.py#L136

Actually the block above it does that too so I prefer to rewrite our jobs and not include this patch.

**Review by ekohl — CHANGES_REQUESTED** [review]
The last patch (cancel-builds-on-update) was merged upstream: https://github.com/openstack-infra/jenkins-job-builder/commit/eb8a54fa45aa1f20aac1207972c0159bf3b09c1a

That means for 2.0 it won't be needed to include it and we only need to send the schedule-failed-builds plugin patch upstream. @GregSutcliffe since you're the author, mind doing the honors?

**Comment by GregSutcliffe** [comment]
I can try to get that upstream, sure. Added to my todo, still a bit under 100% today ...

**Comment by evgeni** [comment]
#3 rebases ontop of 2.0.2, so we should continue there?

**Comment by evgeni** [comment]
we're not using our forked jjb anymore: https://github.com/theforeman/foreman-infra/pull/492

close? :)

**Comment by ekohl** [comment]
Yep, much easier to just deprecate the entire repo :)

**Comment by mmoll** [comment]
could anybody set this repo to "archived" then? :)

**Pull request closed without merging**